blockchain/pallets/minting/src/lib.rs

```diff
@@ -192,7 +196,16 @@ pub mod pallet {
             IdDatasets::<T>::insert(id, extension_proof);
             NextMintingRewards::<T>::insert(id, who);
 
-            Ok(())
+            // Only first call is free, determined by not having an
+            // entry in .CalledRegisterForMinting. All subsequent calls
+            // will be Pays::Yes
+            if CalledRegisterForMinting::<T>::try_get(id).is_ok() {
```

Comment:
I don't think we need a new storage for this. We can probably use existence in the IdDatasets storage.

Reply:
I agree, this will also cover the remove when the address changes on a register_identity if I am correct.